### ISSUE TITLE
GRIM: Characters showing through sets with OpenGL renderer

### ISSUE THREAD

**Issue by ghost** [residualvm/residualvm#1254]
System Specs:

ResidualVM Release: 0.2.1
OS: Ubuntu GNU/Linux 14.04 LTS
VGA compatible controller: Intel Corporation Mobile 945GM/GMS, 943/940GML Express Integrated Graphics Controller (rev 03)
Renderer: OpenGL

Other System Information:
[SysSpecs.txt](https://github.com/residualvm/residualvm/files/323674/SysSpecs.txt)

Manny and other real-time rendered characters tend to show through pre-rendered sets. You can see the worst instance of this that I have encountered in this screenshot:

![grimglitch](https://cloud.githubusercontent.com/assets/17693001/16197746/df2427ca-36c8-11e6-8533-6128b909085e.png)

**Comment by ChelonianEgghead** [comment]
Seems like this is more serious than I thought... I can't complete the Year 2 elevator puzzle. :(

![grimelglitch](https://cloud.githubusercontent.com/assets/17693001/16203103/3e7ad088-36de-11e6-8c72-87b6ff0e3753.png)

**Comment by somaen** [comment]
Presumably this is with the OpenGL-renderer, do you get similar results with the software renderer (toggle-able from the ingame menu)

Additionally, would you mind trying to modify the following setting in .residualvmrc (under your grim-fandango entry):
use_arb_shaders=false

(If it happens to be false already, try with true).

**Comment by ChelonianEgghead** [comment]
Uh... I don't have a use_arb_shaders setting...

Anyway, the elevator puzzle works with software rendering, although it would be nice if the OpenGL renderer worked, since the software one uses considerably more RAM than the former. But then, RAM is not much of a problem if you're one of the 6,999,999 people who has a computer that was made less than ten years ago. ;)

**Comment by ChelonianEgghead** [comment]
Wait, the sets are _pre-rendered_, right?  Seems like I goofed on the title of my first bug report...

**Comment by aquadran** [comment]
as @somaen wrote, try use use_arb_shaders option

**Comment by ChelonianEgghead** [comment]
OK... it seems that I don't have any of the settings mentioned in section 5.2 of residualvm's [readme](https://github.com/residualvm/residualvm/blob/master/README.md) file in my .residualvmrc file. Is this even how you're supposed to do a bug report? I think I should have put this on a forum...

**Comment by aquadran** [comment]
just add use_arb_shaders=false or use_arb_shaders=true to config file

**Comment by ChelonianEgghead** [comment]
Alright, I did that, and here's an image with a side-by-side comparison of the two use_arb_shaders variables taken at the Casino Calavera:
![grim-cn-arb-vs](https://cloud.githubusercontent.com/assets/17693001/16236164/fd9354ea-379c-11e6-928f-896a02e61b95.png)

**Comment by aquadran** [comment]
on both it's drawing fine

**Comment by somaen** [comment]
Did you take these with the software renderer or opengl?

Den tirsdag 21. juni 2016 skrev Paweł Kołodziejski <notifications@github.com>
følgende:

**Comment by ChelonianEgghead** [comment]
Sorry, guys, I was rather mixed up when I closed this issue; I somehow got it in my head that that the settings I was using at the time I closed this were the defaults, other people using the defaults wouldn't be affected, and that someone else who wasn't using the defaults would eventually come along and fix this. WRONG! WRONG! WRONG! Anyway, those last two comparison screenshots were taken using the software renderer, and it doesn't make any difference whether use_arb_shaders=true or false in either rendering mode, so it still doesn't work with OpenGL rendering. In short, this is still a problem for people who just stick with the defaults.
